libraries/apollo-tooling/src/main/resources/meta-introspection.graphql

```diff
@@ -0,0 +1,15 @@
+# Meta introspection query allowing to retrieve server capabilities.
+
+query MetaIntrospectionQuery {
+  __schema {
+    types {
+      name
+      fields {
+        name
+        args {
+          name
+        }
+      }
+    }
+  }
```

Comment:
Download only the required types?

```graphql
query MetaIntrospectionQuery {
  __type(name: "__Schema") { ...typeDetails }
  __type(name: "__Type") { ...typeDetails }
  __type(name: "__Directive") { ...typeDetails }
  __type(name: "__Field") { ...typeDetails }
  __type(name: "__InputValue") { ...typeDetails }
}
```

It's not so much about the performance improvements but the fact that it conveys better what we're going to do with this data?

Reply:
Didn't realize it was allowed! Will update 👍